pkg/cloud/services/s3/s3.go

```diff
@@ -97,43 +98,46 @@ func (s *Service) DeleteBucket() error {
 
 	log.Info("Deleting S3 Bucket")
 
-	// Delete machine pool user data files that did not get deleted
-	// yet by the lifecycle policy
-	for {
-		log.Info("Listing S3 objects of machine pools")
+	if feature.Gates.Enabled(feature.MachinePool) {
```

Comment:
I only found usages of the `feature` package in the controllers package, or in the `main` file. It looks like we are introducing a new dependency between this package and the `feature` package. I'm not sure if this will get accepted upstream, as it seems it goes in a different design direction. I don't know an alternative approach though.

Reply:
Since this code runs in the same process, I think it should work functionality-wise. Regarding upstream, that's the risk we take when implementing in the fork first, so we probably need to accept it this time.